homework/src/index.js

```diff
@@ -29,19 +32,119 @@
+      text: repository.description
+    });
+    const tr3 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr3, {
+      text: 'Forks:'
+    });
+    createAndAppend('td', tr3, {
+      text: repository.forks
+    });
+    const tr4 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr4, {
+      text: 'Updated:'
+    });
+    createAndAppend('td', tr4, {
+      text: repository.updated_at
+    });
+  }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
+  function renderContributors(contributors, container) {
+    container.innerHTML = '';
+    createAndAppend('p', container, {
+      text: 'Contributions'
     });
+    const ul = createAndAppend('ul', container);
+    contributors.forEach(contributor => {
+      const li = createAndAppend('li', ul);
+      const a = createAndAppend('a', li, {
+        href: contributor.html_url,
+        target: '_blank',
+        text: contributor.login
+      });
+      createAndAppend('img', a, {
+        src: contributor.avatar_url
+      });
+      createAndAppend('div', a, {
+        text: contributor.contributions
+      });
+    });
+  }
+
+  function main2(err, repositories) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('div', root, {
+      id: 'header'
+    });
+    createAndAppend('label', header, {
+      text: 'HYF Repositories'
+    });
+    const select = createAndAppend('select', header);
+    if (err) {
+      createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+
+    } else {
+      repositories.sort((a, b) => a.name.localeCompare(b.name));
+      const leftHandContainer = createAndAppend('div', root, { id: 'leftHand' });
+      const rightHandContainer = createAndAppend('div', root, { id: 'rightHand' });
+
+      repositories.forEach((repository, index) => {
+        createAndAppend('option', select, {
+          text: repository.name,
+          value: index
+        });
+      });
+      select.addEventListener('change', () => {
+        const index = select.value;                     //difference between select value and repo index ?
+        const repository = repositories[index];
+        renderRepositories(repository, leftHandContainer);
+        fetchJSON(repository.contributors_url)
+          .then((contributors) => {
+            renderContributors(contributors, rightHandContainer);
+          })
+          .catch((err) => main2(err));
+
+      });
+      renderRepositories(repositories[0], leftHandContainer);
+
+      fetchJSON(repositories[0].contributors_url)
+        .then((contributors) => {
+          renderContributors(contributors, rightHandContainer);
+        })
+        .catch((err) => main2(err));
```

Comment:
Lines 120-125 and lines 128-134 do almost the exact same thing. Can you think of a function that you could write to remove this duplication? You could pass that function two parameters:

1. The repository to render.
2. The container that acts as the parent element.